### PR TITLE
MCR-1978 Validate Session IP with Netmask

### PR DESCRIPTION
**Pull request by OPNA2608**
Validating the session's last IP against the new IP with configurable netmasks
"MCR.Servlet.Session.NetMask.IPv4" and "MCR.Servlet.Session.NetMask.IPv6".

[Link to jira](https://mycore.atlassian.net/browse/MCR-).

### CODE REVIEW

File: mycore-base/src/main/java/org/mycore/frontend/servlets/MCRServlet.java [hunk 1]
```diff
@@ -87,6 +91,16 @@
     private static final boolean ENABLE_BROWSER_CACHE = MCRConfiguration.instance().getBoolean(
         "MCR.Servlet.BrowserCache.enable", false);
 
+    private static final String SESSION_NETMASK_IPV4_STRING = MCRConfiguration.instance()
+        .getString("MCR.Servlet.Session.NetMask.IPv4", "255.255.255.255");
```
Comment (kkrebs) [inline mycore-base/src/main/java/org/mycore/frontend/servlets/MCRServlet.java:95]: these properties should also be part of mycore.properties and documentated there

File: mycore-base/src/main/java/org/mycore/frontend/servlets/MCRServlet.java [hunk 2]
```diff
@@ -87,6 +91,16 @@
...
+
+    private static final String SESSION_NETMASK_IPV6_STRING = MCRConfiguration.instance()
+        .getString("MCR.Servlet.Session.NetMask.IPv6", "FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF");
```
Comment (kkrebs) [inline mycore-base/src/main/java/org/mycore/frontend/servlets/MCRServlet.java:98]: dito

**Review by kkrebs — CHANGES_REQUESTED** [review]

File: mycore-base/src/main/java/org/mycore/frontend/servlets/MCRServlet.java (hunk 1 above)
Comment (OPNA2608) [reply mycore-base/src/main/java/org/mycore/frontend/servlets/MCRServlet.java:95]: Done.

File: mycore-base/src/main/java/org/mycore/frontend/servlets/MCRServlet.java (hunk 2 above)
Comment (OPNA2608) [reply mycore-base/src/main/java/org/mycore/frontend/servlets/MCRServlet.java:98]: Done².

**Review by kkrebs — APPROVED** [review]

**Pull request merged**